packages/v1-core/src/actions/make-proposals.ts

```diff
@@ -22,78 +22,74 @@ const proposalTypes = {
 	},
 };
 
-type ImplementedProposalTypes = {
+export type ImplementedProposalTypes = {
 	[K in ProposalType]: typeof proposalTypes[K] extends () => never ? never : K
   }[ProposalType];
 
+export type ProposalParamWithDeps<T extends ImplementedProposalTypes> = {
+    type: T;
+    params: NonNullable<Parameters<(typeof proposalTypes)[T]>[0]>;
+    deps: NonNullable<Parameters<(typeof proposalTypes)[T]>[1]>;
+};
+
 export const makeProposals = async <T extends ImplementedProposalTypes>(
-	proposalType: T,
-	proposalParams: Parameters<(typeof proposalTypes)[T]>[0],
-	deps: Parameters<(typeof proposalTypes)[T]>[1],
+	config: Config,
+	proposalParams: ProposalParamWithDeps<T>[]
 ) => {
-	invariant(
-		proposalTypes[proposalType],
-		`Proposal type ${proposalType} not found`,
-	);
-	invariant(proposalParams, "Proposal params are required");
-	invariant(deps, "Deps are required");
+	invariant(config, "Config is required");
+	invariant(proposalParams?.length > 0, "Proposal params are required");
 
-	let proposalsWithSignature: ProposalWithSignature[] = [];
+	const proposals: Array<ProposalWithHash> = [];
```

Comment:
```suggestion
	const proposals: ProposalWithHash[] = [];
```

we seem to stick to that on the rest of the codebase